test/cmd/apply.sh

```diff
@@ -46,18 +46,18 @@ run_kubectl_apply_tests() {
   # Post-Condition: deployment "test-deployment-retainkeys" created
   kube::test::get_object_assert deployments "{{range.items}}{{${id_field:?}}}{{end}}" 'test-deployment-retainkeys'
   # Post-Condition: deployment "test-deployment-retainkeys" has defaulted fields
-  kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]:?}" | grep -q RollingUpdate
-  kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]:?}" | grep -q maxSurge
-  kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]:?}" | grep -q maxUnavailable
-  kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]:?}" | grep -q emptyDir
+  grep -q RollingUpdate <<< kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]:?}"
+  grep -q maxSurge <<< kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]:?}"
+  grep -q maxUnavailable <<< kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]:?}"
+  grep -q emptyDir <<< kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]:?}"
   # Command: apply a deployment "test-deployment-retainkeys" should clear
   # defaulted fields and successfully update the deployment
   [[ "$(kubectl apply -f hack/testdata/retainKeys/deployment/deployment-after.yaml "${kube_flags[@]:?}")" ]]
   # Post-Condition: deployment "test-deployment-retainkeys" has updated fields
-  kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]:?}" | grep -q Recreate
-  ! kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]:?}" | grep -q RollingUpdate
-  kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]:?}" | grep -q hostPath
-  ! kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]:?}" | grep -q emptyDir
+  grep -q Recreate <<< kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]:?}"
+  ! grep -q RollingUpdate <<< kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]:?}"
```

Comment:
is `! grep ... <<< ...` equivalent to `! ... | grep ...`? @BenTheElder

Reply:
no, this is wrong. `<<<` is a here string, it will be passed to stdin. this line:
`! grep -q RollingUpdate <<< kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]:?}"`

is equivilant to:

`! echo "kubectl" | grep -q RollingUpdate get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]:?}"`

Reply:
@liggitt @BenTheElder Updated and changed to:

`grep -q RollingUpdate <<< "$(kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]:?}")"`